frontend/src/settings.js

```diff
@@ -1,6 +1,6 @@
 export const baseUrl =
   window.location.hostname === "localhost"
     ? "http://localhost/api"
-    : `http://${window.location.hostname}/api`;
+    : `https://${window.location.hostname}/api`;
 
 export const version = "0.0.1";
```

Comment:
This code patch looks fine. It updates the URL protocol from "http" to "https" when the script is loaded from a non-localhost domain. This is a good practice as it ensures secure communication over the network. 

However, one thing that can be improved is to separate the baseUrl into a configuration file, which can be changed easily without modifying the source code. Additionally, this patch assumes that the API endpoints are located at "/api". If the API endpoints are located at a different location, this variable should be updated accordingly.